client/my-sites/plan-features-2023-grid/types.ts

```diff
@@ -32,4 +32,14 @@ export type PlanProperties = {
 	availableForPurchase: boolean;
 	current?: boolean;
 	showMonthlyPrice: boolean;
+	planActionOverrides?: PlanActionOverrides;
 };
+
+export interface PlanActionOverrides {
+	loggedIn?: {
+		freePlan?: {
+			callback: () => void;
+			text: string;
+		};
+	};
+}
```

Comment:
Is there reason why we go with a nested complex data structure than a simpler key value, dictionary 🤔

Reply:
What exactly do you have in mind? I'm trying to guess from your comment. "simpler key value, dictionary" could be many things

Something flatter you mean i.e. `PlanActionOverrides.loggedInFreePlan`, or even flatter `loggedInFreePlanActionText`, etc?

I'm ok with any option, as long as it's a little DRY. These feel very much temporary to get us to a package. I think the grid is context-aware more than it needs to be with the actions, and we ought to come back to revisit at some point. There is a [follow-up issue](https://github.com/Automattic/wp-calypso/issues/76662) that I created, but for now, I think we patch in whatever way feels more natural.

Reply:
I meant something like `PlanActionOverrides.loggedInFreePlan` 👍 And #76662 sounds perfect!

Reply:
works for me @jdc91 👍  updated